x-pack/plugins/alerting/server/alerts_client/alerts_client.ts

```diff
@@ -197,6 +203,50 @@ export class AlertsClient<
     return { hits, total };
   }
 
+  public async untrackRuleIdsByIndices(indices: string[], ruleIds: string[]) {
+    const esClient = await this.options.elasticsearchClientPromise;
+    const terms: Array<{ term: Record<string, { value: string }> }> = ruleIds.map((ruleId) => ({
+      term: {
+        [ALERT_RULE_UUID]: { value: ruleId },
+      },
+    }));
+    terms.push({
+      term: {
+        [ALERT_STATUS]: { value: ALERT_STATUS_ACTIVE },
+      },
+    });
+    try {
+      // Retry this updateByQuery up to 3 times to make sure the number of documents
+      // updated equals the number of documents matched
+      for (let retryCount = 0; retryCount < 3; retryCount++) {
+        const response = await esClient.updateByQuery({
+          index: indices,
+          allow_no_indices: true,
+          body: {
+            conflicts: 'proceed',
+            script: {
+              source: `ctx._source['${ALERT_STATUS}'] = '${ALERT_STATUS_UNTRACKED}'`,
```

Comment:
For the index threshold & ES query rule type, the AAD docs are not flattened, so using this script, the resulting doc ends up with:

```
{
  "kibana.alert.status": "untracked",
  "kibana": {
    "alert": {
      "status": "active"
     }
  }
}
```

Is there any way to avoid this?

Reply:
Is it just those two rule types specifically or are there potentially more of them? Is there some kind of flag we can look for to know which rule types aren't flattened?

Reply:
I believe anything written by the lifecycle registry is flattened, anything written by the alerting framework is not. I think you should be able to use the alerts definition registered with the rule type. If `ruleType.alerts.shouldWrite=true` it should be unflattened. If `false` or `undefined`, it should be flattened.

Reply:
Never mind, figured out a way to handle this in Painless

Reply:
Even better!